components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/apikey/ApiKeyAuthenticator.java

```diff
@@ -515,7 +515,12 @@ private String generateAndRetrieveBackendJWTToken(String tokenSignature, JWTInfo
             if (token != null) {
                 endUserToken = (String) token;
                 String[] splitToken = ((String) token).split("\\.");
-                JSONObject payload = new JSONObject(new String(Base64.getUrlDecoder().decode(splitToken[1])));
+                JSONObject payload;
+                if (jwtConfigurationDto.getJwtDecoding().equals("base64url")) {
```

Comment:
```suggestion
                if ("base64url".equals(jwtConfigurationDto.getJwtDecoding())) {
```
Let's change this to avoid NPEs. Also, its better to have a constant for this string value.